osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -70,6 +72,9 @@ public IEnumerable<BeatmapSetInfo> Beatmaps
 
         private readonly List<BeatmapGroup> groups = new List<BeatmapGroup>();
 
+        private Bindable<SelectionRandomType> randomType;
+        private readonly HashSet<BeatmapGroup> seenGroups = new HashSet<BeatmapGroup>();

```

Comment:
The existence of duplicates doesn't matter in the first place given that Except() is used, and shouldn't be added to the list anyway. This is less efficient than using List because .NET internally allocates a HashSet with [the elements in the parameter passed to Except()](https://github.com/Microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs#L787) (so really, you're allocating and filling two hashsets).